.circleci/config.yml

```diff
@@ -257,7 +257,52 @@ jobs:
           root: .
           paths:
             - ui
-
+  
+  puppeteer-e2e-test:
+    docker:
+      - image: circleci/node:12.16.1
+    working_directory: ~/workbench
+    steps:
+      - checkout
+      - run:
+          name: Fetch Submodules
+          command: git submodule update --init --recursive
+      - restore_cache:
+          keys:
+          - ui-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
+          - ui-cache-
+      - run: 
+          working_directory: ~/workbench/e2e
+          command: yarn cache clean && yarn install --verbose --frozen-lockfile
+      - save_cache:
+          paths:
+            - ~/.cache/yarn
+          key: ui-cache-{{ checksum "~/workbench/e2e/yarn.lock" }}
+      - run:
+          working_directory: ~/workbench/e2e
+          name: Install Headless Chrome dependencies
+          command: |
+            sudo apt-get install -yq \
+              gconf-service libasound2 libatk1.0-0 libatk-bridge2.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
+              libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 \
+              libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 \
+              libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates \
+              fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget
+      - run:
+          name: Update environment variables
+          command: |
+            echo 'export USER_NAME=$PUPPETEER_TEST_USER' >> $BASH_ENV
+            echo 'export PASSWORD=$PUPPETEER_TEST_USER_PASSWORD' >> $BASH_ENV
+            echo 'export INVITATION_KEY=$PUPPETEER_TEST_REGISTRATION_KEY' >> $BASH_ENV
+            echo 'export CONTACT_EMAIL=hermione.owner@quality.firecloud.org' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Run e2e tests (Puppeteer and Jest)
+          working_directory: ~/workbench/e2e
+          command: yarn --silent test:ci
+      - store_artifacts:
+          path: e2e/logs
```

Comment:
The `logs` folder is going to be created in a follow-up PR for test results and screenshots.